bears/general/AnnotationBear.py

```diff
@@ -88,66 +95,130 @@ def find_annotation_ranges(self,
             A dictionary containing the various ways to define multi-line
             comments in a language.
         :return:
-            Two tuples first containing a tuple of strings, the second a tuple
-            of comments.
+            Four tuples containing dictionary for singleline strings,
+            multiline strings, singleline comments and multiline comments
+            respectively.
         """
         text = ''.join(file)
-        strings_range = []
-        comments_range = []
+        singleline_string_range = []
+        multiline_string_range = []
+        singleline_comment_range = []
+        multiline_comment_range = []
         position = 0
+
         while position <= len(text):
 
             def get_new_position():
-                _range, end_position = self.get_range_end_position(
-                    file,
-                    filename,
-                    text,
-                    multiline_string_delimiters,
-                    position,
-                    self.get_multiline)
-                if end_position and _range:
-                    strings_range.append(_range)
+                end_position, start_delim, end_delim = \
+                    self.get_range_end_position(
+                        file,
+                        filename,
+                        text,
+                        multiline_string_delimiters,
+                        position,
+                        self.get_multiline)
+                if end_position:
+                    seperate_ranges = get_seperate_ranges(file,
+                                                          filename,
+                                                          start_delim,
+                                                          end_delim,
+                                                          position,
+                                                          end_position)
+                    multiline_string = {}
+                    multiline_string[
+                        'start_delimiter_range'] = seperate_ranges[0]
+                    multiline_string[
+                        'end_delimiter_range'] = seperate_ranges[1]
+                    multiline_string['content_range'] = seperate_ranges[2]
+                    multiline_string['full_range'] = seperate_ranges[3]
+                    multiline_string_range.append(multiline_string)
+
                     return end_position + 1
 
-                _range, end_position = self.get_range_end_position(
-                    file,
-                    filename,
-                    text,
-                    string_delimiters,
-                    position,
-                    self.get_singleline_strings)
-                if end_position and _range:
-                    strings_range.append(_range)
+                end_position, start_delim, end_delim = \
```

Comment:
Please do not use line continuation. `\`. Use round brackets.
Also found below.